src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GCStaticsPreInitDataNode.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+using Internal.Text;
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Contains all GC static fields for a particular EEType.
+    /// Fields that have preinitialized data are pointer reloc pointing to frozen objects.
+    /// Other fields are initialized with 0.
+    /// We simply memcpy these over the GC static EEType object.
+    /// </summary>
+    public class GCStaticsPreInitDataNode : ObjectNode, IExportableSymbolNode
+    {
+        private MetadataType _type;
+        private List<PreInitFieldInfo> _sortedPreInitFields;
```

Comment:
Sounds reasonable but this change is getting pretty big as it is. I'll address this in my next change.